utils/wrappers/packing.go

```diff
@@ -142,6 +144,42 @@ func (p *Packer) UnpackLong() uint64 {
 	return val
 }
 
+func (p *Packer) PackUvarInt(v uint64) {
+	buf := binary.AppendUvarint(nil, v)
+	p.PackFixedBytes(buf)
+}
+
+func (p *Packer) UnpackUvarInt() uint64 {
+	p.checkSpace(MinVarintLen)
+	if p.Errored() {
+		return 0
+	}
+	val, bytesRead := binary.Uvarint(p.Bytes)
+	if bytesRead <= 0 {
+		p.Add(errInvalidInput)
+	}
+	p.Bytes = p.Bytes[bytesRead:]
```

Comment:
I think that you're absolutely right ! We definitely should not merge this before solving this issue. It also doesn't really make sense to read the whole `p.Bytes` buffer, but I'm not sure on what to do here because we don't know how much bytes to decode

Reply:
According to:
```
n == 0: buf too small
n  < 0: value larger than 64 bits (overflow)
        and -n is the number of bytes read
```
Seems if `bytesRead` is negative, we should advance the buffer by `-bytesRead`, so perhaps:
```suggestion
	if bytesRead <= 0 {
		p.Add(errInvalidInput)
		bytesRead = -bytesRead // number of bytes read is returned negative in case of overflow
	}
```

Reply:
If the value that is read from the buffer is overflowing 64 bits or that the buffer is too small, do we really still want to read from it ?

Reply:
it also seems acceptable to mark an error and return early instead as you point out